index.js

```diff
@@ -790,29 +871,57 @@ class GitlabScm extends Scm {
      * @param  {String}   config.token      The token used to authenticate to the SCM
      * @return {Promise}
      */
-    async _addPrComment({ comment, prNum, scmUri, token }) {
+    async _addPrComment({ comment, prNum, scmUri }) {
         const { repoId } = getScmUriParts(scmUri);
 
-        return this.breaker.runCommand({
-            json: true,
-            method: 'POST',
-            auth: {
-                bearer: token
-            },
-            url: `${this.config.gitlabProtocol}://${this.config.gitlabHost}/api/v4` +
-                 `/projects/${repoId}/merge_requests/${prNum}/notes`,
-            qs: {
-                body: comment
+        const prComments = await this.prComments(repoId, prNum);
+
+        if (prComments) {
+            const botComment = prComments.comments.find(commentObj =>
+                commentObj.author.username === this.config.username);
+
+            if (botComment) {
+                try {
+                    const pullRequestComment = await this.editPrComment(
+                        botComment.id, repoId, prNum, comment);
+
+                    return {
+                        commentId: `${pullRequestComment.body.id}`,
+                        createTime: `${pullRequestComment.body.created_at}`,
+                        username: pullRequestComment.body.author.username
+                    };
+                } catch (err) {
+                    logger.error('Failed to addPRComment: ', err);
+
+                    return null;
+                }
             }
-        }).then((response) => {
-            checkResponseError(response, '_addPrComment');
+        }
+
+        try {
+            const pullRequestComment = await this.breaker.runCommand({
+                json: true,
+                method: 'POST',
+                auth: {
+                    bearer: this.config.commentUserToken
+                },
+                url: `${this.config.gitlabProtocol}://${this.config.gitlabHost}/api/v4` +
+                     `/projects/${repoId}/merge_requests/${prNum}/notes`,
+                qs: {
+                    body: comment
+                }
+            });
```

Comment:
if it is convenient, refactor the api calls to a single func with url as param

Reply:
will update at a different time